src/Mod/Sketcher/Gui/SnapManager.cpp

```diff
@@ -0,0 +1,309 @@
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+#include <QApplication>
+#endif  // #ifndef _PreComp_
+
+#include <Mod/Sketcher/App/SketchObject.h>
+
+#include "SnapManager.h"
+#include "ViewProviderSketch.h"
+
+
+using namespace SketcherGui;
+using namespace Sketcher;
+
+/************************************ Attorney *******************************************/
+
+inline int ViewProviderSketchSnapAttorney::getPreselectPoint(const ViewProviderSketch& vp)
+{
+    return vp.getPreselectPoint();
+}
+
+inline int ViewProviderSketchSnapAttorney::getPreselectCross(const ViewProviderSketch& vp)
+{
+    return vp.getPreselectCross();
+}
+
+inline int ViewProviderSketchSnapAttorney::getPreselectCurve(const ViewProviderSketch& vp)
+{
+    return vp.getPreselectCurve();
+}
+
+/**************************** ParameterObserver nested class *****************************/
+SnapManager::ParameterObserver::ParameterObserver(SnapManager& client) : Client(client)
+{
+    initParameters();
+    subscribeToParameters();
+}
+
+SnapManager::ParameterObserver::~ParameterObserver()
+{
+    unsubscribeToParameters();
+}
+
+void SnapManager::ParameterObserver::initParameters()
+{
+    // static map to avoid substantial if/else branching
+    //
+    // key->first               => String of parameter,
+    // key->second              => Update function to be called for the parameter,
+    str2updatefunction = {
+        {"SnapToObjects",
+            [this](const std::string& param) {updateSnapToObjectParameter(param); }},
+        {"SnapAngle",
+            [this](const std::string& param) {updateSnapAngleParameters(param); }},
+    };
+
+    for (auto& val : str2updatefunction) {
+        auto string = val.first;
+        auto function = val.second;
+
+        function(string);
+    }
+}
+
+void SnapManager::ParameterObserver::updateSnapToObjectParameter(const std::string& parametername)
+{
+    (void)parametername;

```

Comment:
Instead of voiding the parameter name as unused, I would used it in getBool (instead of hardcoding it there). The reason is that this code will likely be refactor when creating the object I refer to in the next paragraph. Not hardcoding the parameter would help that effort.

Ideally, all parameters would use a single update function that is a template function and can select the getXXX from it. Because this kind of parameter observer is being used often, I am starting to consider subclassing ParameterObserver, so that we can get the extra function in way fewer lines. But that is a different fight for a different PR.

Reply:
I was about to add the same remark yesterday, but as there are only 2 parameters of 2 different types there, I feel like it isn't mandatory now. :rofl: But indeed it's not versatile as an implementation. ;)